### DISCUSSION TITLE
Welcome to VQGAN-CLIP Discussions!

### DISCUSSION THREAD

**Discussion by nerdyrodent** [nerdyrodent/VQGAN-CLIP#36] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by constantupgrade** [comment]
Hi, I have a couple questions. 

Firstly, how do I get it to save every x amount of frames? I used the --save_every flag and it just saves over the same image. i was wondering if it's possible to save them out to new images each time.

Secondly, the random.sh doesn't work for me, I get an error saying "no module called tqdm". This could probably be a number of things and I don't really know what info to give so i understand if theres no response to this question.

I am on Windows 10 if that changes anything.

Thanks for all your time and effort.

**Reply by jonnm4** [reply]
-se only updates your "working image" you still have to copy it somewhere else and rename it. Or, yes. It just get rewritten. Personally, I use -i 50; then copy the file some where, then use your "working image" as the init_image for the next frame -ii "Working_Img.png" -i 50 to start up again from where you left off. try not to go lower then 20 or I find it doesn't learn very much.

**Comment by StellaAthena** [comment]
Are you in the EleutherAI discord server? I'm looking to create an official local implementation and I was thinking of basing it off of a mixture of yours and BATbot. Definitely want to talk about it though + assign proper credit.

**Comment by FrenchCroissant** [comment]
Hi there, sorry to bother you guys. I'm a starting python learner and I'm wondering if there is someone who could help me with the vqgan-clip. I've been searching a lot for this but it seems that there is no one else interested in what I would like to do.

I want to make the gan loop through a batch of images (purposed for the init_image and target_images) I’ve stored in a directory. And along with that I want to use a set of preselected text prompts and randomly choose 2 of those text prompts each run in the loop.

Hope someone can help me and thank you in advance! (btw: Katherine Crowson's VQGAN+CLIP on google Colab)

**Reply by jonnm4** [reply]
I'm not sure how to do it from a google Colab notebook but if you run it locally with nerdyrodent's code then it's just a matter of writting a bash script to pick 2 random lines from a "Prompts.txt" file and your init_image. keep track of what init_image were on `padded_count=$(printf "%04d" "$i");` then just generate image `python generate.py -p "$Prompt1"|"$Prompt2" -ii image-"$padded_count".png;` and of coarse increment "i" so we go to the next image, next time `$(++i)`
